### ISSUE TITLE
 No OpKernel was registered to support Op 'Maximum' with these attrs.---iOS

### ISSUE THREAD

**Issue by StarRain-L** [tensorflow/tensorflow#10677]
Consult: excuse me, write own model file, called collapse, iOS,

![snip20170613_6](https://user-images.githubusercontent.com/8908244/27081496-407ad54a-5073-11e7-8464-faab2ca4836e.png)

**Comment by aselle** [comment]
It looks like you are invoking something involving a gradient. If you are training, you likely need to change the makefile to add more ops to the build. Correct @petewarden?

**Comment by StarRain-L** [comment]
Thank you for your reply. I'm not quite understand what you say, and how to change the makefile to add more ops to the build?please!!!

**Comment by aselle** [comment]
My point is that you appear to trying to run the training part of your graph rather than the inference part? Is this correct? The set of ops included in ios is limited to training, so you'll have to play with the makefile to add more ops to the build. See:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/makefile/README.md
and search for other issues. I think your issue is a duplicate of
https://github.com/tensorflow/tensorflow/issues/2929

**Comment by StarRain-L** [comment]
Thank you for your reply, and I have read "READEME"many times. I don't know how to add more ops to the build.I wants to use his own model.My app built for Mobile will crash at runtime. I provide a screenshot.On the left side of the picture is the source code,and the path is "tensorflow/core/ops/math_ops.cc".On the right is my crash logs. I found that "DT_FLOAT" did not found in the source.How to solve? 
![27174808-ed5e7b54-51ef-11e7-8813-a94590972702](https://user-images.githubusercontent.com/8908244/27207657-283096da-5273-11e7-9315-9750a91c8f25.png)

**Comment by aselle** [comment]
Are you trying to do inference or training? If you are just trying to do inference, then you should really find out why your session is invoking ops that are called "gradients/...". It is far easier to figure out why that is happening than to add ops, and it probably will be important for performance.

**Comment by ccwinner** [comment]
@aselle  Item seems @StarRain-L is trying to use his own model written in python, it works well before. Your referenced issue has been closed and related missed ops has already been added in tf_op_files.txt in TF folder.

**Comment by StarRain-L** [comment]
@aselle  I am doing the inference , and do you mean the session should not invoking ops  "gradients" when doing the inference ? how can I remove these ops ? I just use the model file saved by python .

**Comment by aselle** [comment]
Can you give example code that exhibits the problem. Essentially when you call session.run() you ask for a given op and every dependent of that has to be run. You have a dependency on something called "Gradients/..." (that's where the error is showing up). I feel like you might be pulling on a training op or something dependent on a training op. If not, it is somewhat strange. Either way you could work to modify the build to add ops, but it is quite difficult.

**Comment by aselle** [comment]
Also check out this for more info... https://codelabs.developers.google.com/codelabs/tensorflow-for-poets-2/#0

**Comment by StarRain-L** [comment]
@aselle Attach the code , and please have a look.how can I modify these ops ?
![ab8a4371-8743-40ed-b960-5147afd686b2](https://user-images.githubusercontent.com/8908244/27252215-fa468072-538a-11e7-8ee6-24ad41c08aed.png)

**Comment by aselle** [comment]
Can you detail how you are creating your model, how you are constructing the model (what framework), the commands and programs you used to compact the model and freeze it. It seems what is happening is that a part of the training network is brought in. If all else fails, you could try to construct an inference only network (that uses the same variables so you can load your trained checkpoint in it).

**Comment by StarRain-L** [comment]
Very grateful for your reply. I understand a little bit about, I try again!!

**Comment by StarRain-L** [comment]
@aselle The problem is resolved.My colleague reference your instructions to modify the model code, I am not very clear what he modified.And  tf_op_files.txt increased " random_op. cc".I provide the reference link.Finally, thank you very much for your help, and if have any problem, I hope I can continue to ask you.
> https://github.com/tensorflow/tensorflow/search?l=C%2B%2B&q=TruncatedNormal&type=&utf8=✓

https://stackoverflow.com/questions/40855271/no-opkernel-was-registered-to-support-op-switch-with-these-attrs-on-ios/43627334

**Comment by tatatodd** [comment]
Closing this out, as it looks like the problem was resolved.
